src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -78,6 +78,14 @@ private function completeDefinition($id, Definition $definition)
 
             try {
                 if (!$typeHint = $parameter->getClass()) {
+                    // no default value? Then fail
+                    if (!$parameter->isOptional()) {
+                        throw new RuntimeException(sprintf('Unable to autowire argument index %s ($%s) for the service "%s". If this is an object, give it a type-hint. Otherwise, specify this argument\'s value explicitly.', $index, $parameter->name, $id));
+                    }
+
+                    // specifically pass the default value
+                    $arguments[$index] = $parameter->getDefaultValue();
```

Comment:
We can do this, but it's tricky. For example:

```php
public function __construct($foo = 'foo', LoggerInterface $logger = null)
```

In this case, `$foo` is optional. If we omit it but then `$logger` becomes mapped because of autowiring, we suddenly *will* need to pass the `$foo` value. This means we'll need to not set optional values here, but keep track of them and add them at the end if we detect that they are suddenly needed. Does that make sense? I was waiting to see if anyone asked about this.